Tests/SwiftBSONTests/BSONTests.swift

```diff
@@ -6,19 +6,15 @@ import SwiftBSON
 import XCTest
 
 open class BSONTestCase: XCTestCase {
-    /// Gets the path of the directory containing spec files, depending on whether
-    /// we're running from XCode or the command line
+    /// Gets the path of the directory containing spec files.
     static var specsPath: String {
-        // if we can access the "/Tests" directory, assume we're running from command line
-        if FileManager.default.fileExists(atPath: "./Tests") {
-            return "./Tests/Specs"
-        }
-        // otherwise we're in Xcode, get the bundle's resource path
-        guard let path = Bundle(for: self).resourcePath else {
-            XCTFail("Missing resource path")
-            return ""
-        }
-        return path
+        // Approach taken from https://stackoverflow.com/a/58034307
+        // TODO: SWIFT-1442 Once we drop Swift < 5.3 we can switch to including the JSON files as Resources via our
+        // package manifest instead.
+        let thisFile = URL(fileURLWithPath: #file)
```

Comment:
what is `#file` here?

Reply:
it gives the full path to the file it appears in - so in this case it would give the full path to `Tests/SwiftBSONTests/BSONTests.swift`.